src/main/java/server/ServerHandler.java

```diff
@@ -65,12 +77,9 @@ private void executeCommand(Message message, Session mainSession) {
         try {
             switch (command) {
                 case "/list":
-                    for (SelectionKey key : selector.keys()) {
-                        if (key.isValid() && key.channel() instanceof SocketChannel) {
-                            Session session = (Session) key.attachment();
-                            Message tempMessage = new Message(MessageType.MESSAGE, "[SERVER]", session.getUser().getName());
-                            mainSession.writeMessage(tempMessage);
-                        }
+                    for (Session session : sessionSet) {
+                        Message tempMessage = new Message(MessageType.MESSAGE, "[SERVER]", session.getUser().getName());
+                        mainSession.writeMessage(tempMessage);
```

Comment:
Why do we send each user in a separate message instead of aggregating them into one message?